Sources/Models/AppsResponse.swift

```diff
@@ -30,43 +30,88 @@ public struct AppsResponse: Decodable {
-            }
-            
-            if let wrapped = try? BetaLicenseAgreement(from: decoder) {
-                self = .betaLicenseAgreement(wrapped)
-                return
-            }
-            
-            if let wrapped = try? BetaAppReviewDetail(from: decoder) {
-                self = .betaAppReviewDetail(wrapped)
-                return
+            let type = try decoder.container(keyedBy: TypeCodingKeys.self).decode(String.self, forKey: .type)
+            switch type {
+            case "betaGroups":
+                self = try .betaGroup(BetaGroup(from: decoder))
+            case "preReleaseVersions":
+                self = try .prereleaseVersion(PrereleaseVersion(from: decoder))
+            case "betaAppLocalizations":
+                self = try .betaAppLocalization(BetaAppLocalization(from: decoder))
+            case "builds":
+                self = try .build(Build(from: decoder))
+            case "betaLicenseAgreements":
+                self = try .betaLicenseAgreement(BetaLicenseAgreement(from: decoder))
+            case "betaAppReviewDetails":
+                self = try .betaAppReviewDetail(BetaAppReviewDetail(from: decoder))
+            default:
+                print("\(type)")
+                throw DecodingError.typeMismatch(
+                    Included.self,
+                    DecodingError.Context(codingPath: [], debugDescription: "Not convertable to \(Included.self)")
+                )
             }
-            
-            throw DecodingError.typeMismatch(
-                Included.self,
-                DecodingError.Context(codingPath: [], debugDescription: "Not convertable to \(Included.self)")
-            )
+        }
+    }
+}
+
+extension AppsResponse.Included {
+    
+    public var betaGroup: BetaGroup? {
+        switch self {
+        case .betaGroup(let value):
+            return value
+        default:
+            return nil
+        }
+    }
+    
+    public var prereleaseVersion: PrereleaseVersion? {
+        switch self {
+        case .prereleaseVersion(let value):
+            return value
+        default:
+            return nil
+        }
+    }
+    
+    public var betaAppLocalization: BetaAppLocalization? {
+        switch self {
+        case .betaAppLocalization(let value):
+            return value
+        default:
+            return nil
+        }
+    }
+    
+    public var build: Build? {
+        switch self {
+        case .build(let value):
+            return value
+        default:
+            return nil
+        }
+    }
+    
+    public var betaLicenseAgreement: BetaLicenseAgreement? {
+        switch self {
+        case .betaLicenseAgreement(let value):
+            return value
+        default:
+            return nil
+        }
+    }
+    
+    public var betaAppReviewDetail: BetaAppReviewDetail? {
+        switch self {
+        case .betaAppReviewDetail(let value):
+            return value
+        default:
+            return nil
```

Comment:
We're creating quite some duplicate code here. Would be awesome if we can reuse logic across models!